gc/base/HeapVirtualMemory.cpp

```diff
@@ -312,7 +312,14 @@ MM_HeapVirtualMemory::commitMemory(void* address, uintptr_t size)
 {
 	MM_GCExtensionsBase* extensions = MM_GCExtensionsBase::getExtensions(_omrVM);
 	MM_MemoryManager* memoryManager = extensions->memoryManager;
-	return memoryManager->commitMemory(&_vmemHandle, address, size);
+
+	bool resultCommitMemory=memoryManager->commitMemory(&_vmemHandle, address, size);
+
+	if (resultCommitMemory && extensions->pretouchHeapOnExpand) {
+		memset(address, 0, size);
```

Comment:
There is no need to zero entire memory. There is enough to touch one byte per physical page. Physical page size can be taken from `_vmemHandle`

Reply:
@amicic Do we really want to pre-zero heap memory or just need to get all heap related physical memory pages from OS?

Reply:
the latter, but I don't think that it will be much of difference (in speed) if we go and just memset all of it with 0.